File: files/2025-08-10/CSEC 10082025.R
library(tidyverse)
library(readxl)

path = "files/2025-08-10/Challenge 50.xlsx"
input = read_excel(path, range = "A2:C14")
test  = read_excel(path, range = "E2:E6")

param = "National ID"

result = input %>%
  nest_by(`Staff No.`) %>%
  mutate(has_id = any(data$Identifiers == param)) %>%
  filter(!has_id | is.na(has_id)) %>%
  select(`Staff No.`)

all.equal(result$`Staff No.`, test$`Staff Without National ID`)
# > [1] TRUE

File: files/2025-08-10/CSEC 10082025.py
import pandas as pd

path = "files/2025-08-10/Challenge 50.xlsx"
param = "National ID"
input = pd.read_excel(path, usecols="A:C", skiprows=1, nrows = 12)
test  = pd.read_excel(path, usecols="E", skiprows=1, nrows = 4).squeeze().to_list()

result = (input.groupby('Staff No.')
            .filter(lambda g: param not in g['Identifiers'].values)
            ['Staff No.'].unique().tolist())

print(result == test) # True